plugins/modules/monitoring/airbrake_deployment.py

```diff
@@ -74,44 +94,80 @@ def main():
 
     module = AnsibleModule(
         argument_spec=dict(
-            token=dict(required=True, no_log=True),
+            token=dict(required=False, no_log=True),
+            project_id=dict(required=False, no_log=True),
+            project_key=dict(required=False, no_log=True),
             environment=dict(required=True),
             user=dict(required=False),
             repo=dict(required=False),
             revision=dict(required=False),
-            url=dict(required=False, default='https://api.airbrake.io/deploys.txt'),
+            url=dict(required=False, default='https://api.airbrake.io/api/v4/projects/'),
             validate_certs=dict(default='yes', type='bool'),
         ),
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_together=[('project_id', 'project_key')],
+        mutually_exclusive=[('project_id', 'token')],
     )
 
-    # build list of params
+    # Build list of params
     params = {}
 
-    if module.params["environment"]:
-        params["deploy[rails_env]"] = module.params["environment"]
+    # If we're in check mode, just exit pretending like we succeeded
+    if module.check_mode:
+        module.exit_json(changed=True)
 
-    if module.params["user"]:
-        params["deploy[local_username]"] = module.params["user"]
+    if module.params["token"]:
+        if module.params["environment"]:
+            params["deploy[rails_env]"] = module.params["environment"]
 
-    if module.params["repo"]:
-        params["deploy[scm_repository]"] = module.params["repo"]
+        if module.params["user"]:
+            params["deploy[local_username]"] = module.params["user"]
 
-    if module.params["revision"]:
-        params["deploy[scm_revision]"] = module.params["revision"]
+        if module.params["repo"]:
+            params["deploy[scm_repository]"] = module.params["repo"]
 
-    params["api_key"] = module.params["token"]
+        if module.params["revision"]:
+            params["deploy[scm_revision]"] = module.params["revision"]
 
-    url = module.params.get('url')
+        module.deprecate("Parameter 'token' is deprecated in 2.10. Please remove it and use 'project_id' and 'project_key' instead", version='2.12')
```

Comment:
Please keep 2.14 instead of 2.12. If the sanity tests complain, add a ignore.txt entry for it.